planner/codec/codec.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codec
```

Comment:
how about moving these code to the `util` package?

Reply:
Move pkg to `util` pkg maybe make the `util` pkg complexity and unwieldy,
And try to put `encodePlan` and `decodePlan` in the same directory is better?